src/jwe.c

```diff
@@ -794,6 +795,22 @@ static int _r_concat_kdf(jwe_t * jwe, jwa_alg alg, const gnutls_datum_t * Z, gnu
     kdf->data[kdf->size+3] = (unsigned char)(key_data_len) & 0xFF;
     kdf->size += 4;
 
+    size_t derived_key_len = _r_get_key_size(jwe->enc);
```

Comment:
for better readability, I prefer to declare all variables in the beginning of the function

Reply:
this is fair request, sure will update the PR accordingly